src/libstore/store-api.cc

```diff
@@ -200,9 +200,9 @@ StorePath Store::makeTextPath(std::string_view name, const Hash & hash,
 
 
 std::pair<StorePath, Hash> Store::computeStorePathForPath(std::string_view name,
-    const Path & srcPath, bool recursive, HashType hashAlgo, PathFilter & filter) const
+    const Path & srcPath, FileIngestionMethod recursive, HashType hashAlgo, PathFilter & filter) const
 {
-    Hash h = recursive ? hashPath(hashAlgo, srcPath, filter).first : hashFile(hashAlgo, srcPath);
+    Hash h = static_cast<bool>(recursive) ? hashPath(hashAlgo, srcPath, filter).first : hashFile(hashAlgo, srcPath);
```

Comment:
`static_cast<bool>()` seems to negate the advantage of using an enum, since it requires knowing that `Recursive` casts to `true`. Maybe better to be explicit (`method == Recursive`).

Also the variable name `recursive` should be something like `ingestionMethod`.

Reply:
I thought changing the enum "superclass" would break this, making it easier to fix later, but it doesn't. I'll definitely fix now.